FDK17プロジェクト/コード/03.サウンド/CSound.cs

```diff
@@ -728,39 +807,25 @@ public CSound()
 			this._hTempoStream = 0;
 		}
 
-		public object Clone()
-		{
-			if ( !bDirectSoundである )
-			{
-				throw new NotImplementedException();
-			}
-			CSound clone = (CSound) MemberwiseClone();	// これだけだとCY連打が途切れる＆タイトルに戻る際にNullRef例外発生
-			this.DirectSound.DuplicateSoundBuffer( this.Buffer, out clone.Buffer );
-
-			// CSound.listインスタンス.Add( this );			// インスタンスリストに登録。
-			// 本来これを加えるべきだが、Add後Removeできなくなっている。Clone()の仕方の問題であろう。
-
-			return clone;
-		}
 		public void tASIOサウンドを作成する( string strファイル名, int hMixer )
 		{
+		    this.eデバイス種別 = ESoundDeviceType.ASIO;		// 作成後に設定する。（作成に失敗してると例外発出されてここは実行されない）
```

Comment:
Here and below we set the sound device type slightly earlier than the previous code, so that important conditional code will run inside tBASSサウンドを作成する(...)